BaragonData/src/main/java/com/hubspot/baragon/data/BaragonStateDatastore.java

```diff
@@ -99,11 +101,18 @@ public void updateService(BaragonRequest request) throws Exception {
     String serviceId = request.getLoadBalancerService().getServiceId();
     Collection<UpstreamInfo> currentUpstreams = getUpstreams(serviceId);
     String servicePath = String.format(SERVICE_FORMAT, serviceId);
-    CuratorTransactionFinal transaction;
-    if (nodeExists(servicePath)) {
-      transaction = curatorFramework.inTransaction().setData().forPath(servicePath, serialize(request.getLoadBalancerService())).and();
-    } else {
-      transaction = curatorFramework.inTransaction().create().forPath(servicePath, serialize(request.getLoadBalancerService())).and();
+
+    if (!request.isNoServiceUpdate() && !nodeExists(servicePath)) {
+      throw new RuntimeException("BaragonRequest " + request.getLoadBalancerRequestId() + " has noServiceUpdate=true, but no BaragonService exists!");  // TODO: better exception?
```

Comment:
Should we be smart enough to handle this case in Baragon maybe? fFor example if deploying a new service for hte first time and the one with noServiceUpdate gets done and ready forLB first, wouldn't we want to just add it?